skyportal/handlers/api/source.py

```diff
@@ -1366,6 +1389,56 @@ class Validator(Schema):
                     ) = result
                 else:
                     (obj,) = result
+
+                if annotations_filter is not None:
+                    annotations = (
+                        Annotation.query_records_accessible_by(self.current_user)
+                        .filter(Annotation.obj_id == obj.id)
+                        .all()
+                    )
+
+                    passes_filter = True
+                    for ann_filt in annotations_filter:
+                        ann_split = ann_filt.split(":")
+                        if not (len(ann_split) == 1 or len(ann_split) == 3):
+                            return self.error(
+                                "Invalid annotationsFilter value -- annotation filter must have 1 or 3 values"
+                            )
+                        name = ann_split[0].strip()
+                        if len(ann_split) == 3:
+                            value = ann_split[1].strip()
+                            try:
+                                value = float(value)
+                            except ValueError as e:
+                                return self.error(
+                                    f"Invalid annotation filter value: {e}"
+                                )
+                            op = ann_split[2].strip()
+                        # first check that the name is present
+                        name_present = [
+                            isinstance(an.data, dict) and name in an.data
+                            for an in annotations
+                        ]
+                        name_check = any(name_present)
+
+                        # fails the filter if name is not present
+                        if not name_check:
+                            passes_filter = False
+                            break
+                        if len(ann_split) == 3:
+                            index = name_present.index(True)
+                            data_value = annotations[index].data[name]
+                            try:
+                                comp_function = eval(f"operator.{op}")
```

Comment:
I'm a little uneasy about this `eval`. Are there really that many operations that we can't just preconstruct a dictionary of keywords -> functions instead?

Reply:
i think i got this one.